dialogflow/context_management.rb

```diff
@@ -0,0 +1,116 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require 'securerandom'
+
+def list_contexts project_id:, session_id:
+  # [START dialogflow_list_contexts]
+  # project_id = "Your Google Cloud project ID"
+  # session_id = "existing_session_id"
+  
+  require "google/cloud/dialogflow"
+
+  contexts_client = Google::Cloud::Dialogflow::Contexts.new
+  session_path = contexts_client.class.session_path project_id, session_id
+
+  contexts = contexts_client.list_contexts(session_path)
+
+  puts "Contexts for session #{session_path}:\n"
+  contexts.each do |context|
+    puts "Context name:   #{context.name}"
+    puts "Lifespan count: #{context.lifespan_count}"
+    if !context.parameters.nil?
+      puts "Fields:"
+      context.parameters.fields.each do |field, value|
+        if value.string_value
+          puts "\t#{field}: #{value.string_value}"
+        end
+      end
+    end
+  end
+  # [END dialogflow_list_contexts]
+end
+
+
+def create_context project_id:, session_id:, context_id:
+  # [START dialogflow_create_context]
+  # project_id = "Your Google Cloud project ID"
+  # session_id = "existing_session_id"
+  # context_id = "new_context_id"
+  
+  require "google/cloud/dialogflow"
+
+  contexts_client = Google::Cloud::Dialogflow::Contexts.new
+  session_path = contexts_client.class.session_path project_id, session_id
```

Comment:
Another option is ignoring the GAPIC path helpers ~ we've had lots of friction from users with these. In the long term, resource path helpers should be baked into the library and hidden from users. In the meantime, consider using a **simple string** instead: `"#{project_id}/#{session_id}"`

```ruby
contexts = contexts_client.list_contexts "#{project_id}/#{session_id}"

puts "Contexts for session #{session_id}:"
context.each do |context|
  puts "Context name:   #{context.name}"
  puts "Lifespan count: #{context.lifespan_count}"
  if context.parameters.any?
    puts "Fields:"
    context.parameters.fields.each do |field, value|
      puts "#{field} #{value.value}"
    end
  end
end
```

Reply:
Having the client library handle this would be ideal.  

The context path template actually looks like this:
```
"projects/{project}/agent/sessions/{session}/contexts/{context}"
```

Compared to having something like this in the samples, I would rather call client.class's helpers.  Additionally I am concerned about doing this in the samples since it could break the samples when down the road the shape of the resource paths change.  For instance if multiple agents are allowed for a project there might be an additional segment between `agent` and `sessions`.